lib/tpm2_options.c

```diff
@@ -279,23 +279,30 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv, char **envp,
      * grep -rn case\ \'[a-zA-Z]\' | awk '{print $3}' | sed s/\'//g | sed s/\://g | sort | uniq | less
      */
     struct option long_options [] = {
-        { "tcti",          required_argument, NULL, 'T' },
         { "help",          no_argument,       NULL, 'h' },
         { "verbose",       no_argument,       NULL, 'v' },
         { "quiet",         no_argument,       NULL, 'Q' },
         { "version",       no_argument,       NULL, 'V' },
         { "enable-errata", no_argument,       NULL, 'Z' },
+        /*
+         * tcti must be last, because when a tool requests that
+         * a sapi is not provided, we subtract 1 from the size
+         * of the array length so --tcti is not valid.
+         */
+        { "tcti",          required_argument, NULL, 'T' },
     };
+    size_t long_options_len =  ARRAY_LEN(long_options);
 
     const char *tcti_conf_option = NULL;
 
     /* handle any options */
     const char* common_short_opts = "T:hvVQZ";
     if (tool_opts && (tool_opts->flags & TPM2_OPTIONS_NO_SAPI)) {
```

Comment:
wonder if we should rename this to `TPM2_OPTIONS_NO_TCTI` ?

Reply:
Well it ultimately ends up with a NULL sapi pointer from the tools perspective